File: check_schema.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()

url = os.environ.get("VITE_SUPABASE_URL")
key = os.environ.get("VITE_SUPABASE_ANON_KEY")

supabase: Client = create_client(url, key)

try:
    # Tenta pegar um registro qualquer para ver as colunas
    response = supabase.table("ef_classes").select("*").limit(1).execute()
    print("Sucesso ao conectar!")
    if response.data:
        print("Colunas encontradas:", response.data[0].keys())
    else:
        print("Tabela vazia. Tentando obter estrutura via query...")
        # Se estiver vazia, não conseguimos ver as colunas via select *
        # Mas podemos tentar inserir um dado de teste com 'id' para ver se falha
except Exception as e:
    print("Erro ao acessar ef_classes:", e)

File: check_schema.js
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
dotenv.config();

const supabaseUrl = process.env.VITE_SUPABASE_URL;
const supabaseKey = process.env.VITE_SUPABASE_ANON_KEY;

const supabase = createClient(supabaseUrl, supabaseKey);

async function check() {
    console.log("Conectando ao Supabase...");
    const { data, error } = await supabase.from('ef_classes').select('*').limit(1);

    if (error) {
        console.error("Erro ao ler ef_classes:", error.message);
        const { error: error2 } = await supabase.from('ef_teachers').select('*').limit(1);
        if (error2) console.error("Erro ao ler ef_teachers:", error2.message);
        else console.log("Conexão com ef_teachers OK.");
    } else {
        console.log("Sucesso! Colunas em ef_classes:", data.length > 0 ? Object.keys(data[0]) : "Tabela vazia (mas conexão OK)");
        // Se estiver vazia, vamos tentar pegar o esquema de outra forma:
        if (data.length === 0) {
            const { data: cols, error: errCols } = await supabase.rpc('get_table_columns', { table_name: 'ef_classes' });
            // Provavelmente RPC não existe, mas vamos ver os erros
        }
    }
}

check();
